core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2602,6 +2611,30 @@ private RelBuilder aggregate_(GroupKeyImpl groupKey,
     return project(projects.transform((i, name) -> aliasMaybe(field(i), name)));
   }
 
+  public static Set<Integer> getAllFields2(ImmutableBitSet groupSet,
+      List<AggregateCall> aggCallList, boolean checkSubquery, int expcnt) {
+    final Set<Integer> allFields = new TreeSet<>();
+    allFields.addAll(groupSet.asList());
+    if (checkSubquery) {
+      for (int i = 0; i < expcnt; i++) {
+        allFields.add(i);
+      }
+
+      return allFields;
+    }
```

Comment:
here as well rename expcnt into a descriptive into something descriptive
Also add comments about the intend of the checkSubquery condition i.e. addition of fields inside the loop.

Reply:
resolved